packages/app/src/background/services/connection/ConnectionService.ts

```diff
@@ -0,0 +1,170 @@
+import { EventName } from "@cryptkeeperzk/providers";
+import omit from "lodash/omit";
+import pick from "lodash/pick";
+
+import BrowserUtils from "@src/background/controllers/browserUtils";
+import { type BackupData, IBackupable } from "@src/background/services/backup";
+import BaseService from "@src/background/services/base";
+import CryptoService, { ECryptMode } from "@src/background/services/crypto";
+import SimpleStorage from "@src/background/services/storage";
+import ZkIdentityService from "@src/background/services/zkIdentity";
+
+import type { IIdenityConnection, IZkMetadata, IConnectArgs, IIdentityMetadata } from "@cryptkeeperzk/types";
+
+const CONNECTIONS_STORAGE_KEY = "@@CONNECTIONS@@";
+
+export default class ConnectionService extends BaseService implements IBackupable {
+  private static INSTANCE?: ConnectionService;
+
+  private connections: Map<string, IIdenityConnection>;
+
+  private readonly connectionsStorage: SimpleStorage;
+
+  private readonly cryptoService: CryptoService;
+
+  private readonly browserController: BrowserUtils;
+
+  private readonly zkIdenityService: ZkIdentityService;
+
+  private constructor() {
+    super();
+    this.connections = new Map();
+    this.connectionsStorage = new SimpleStorage(CONNECTIONS_STORAGE_KEY);
+    this.cryptoService = CryptoService.getInstance();
+    this.browserController = BrowserUtils.getInstance();
+    this.zkIdenityService = ZkIdentityService.getInstance();
+  }
+
+  static getInstance = (): ConnectionService => {
+    if (!ConnectionService.INSTANCE) {
+      ConnectionService.INSTANCE = new ConnectionService();
+    }
+
+    return ConnectionService.INSTANCE;
+  };
+
+  lock = async (): Promise<boolean> => this.onLock(this.onClean);
```

Comment:
Why this `unlock()` and `lock()` is not added with these: 
```ts
// Handling RPC INTERNAL ACTIONS
    // common
    this.handler.add(
      RPCInternalAction.UNLOCK,
      this.lockService.unlock,
      this.approvalService.unlock,
      this.zkIdentityService.unlock,
      this.lockService.onUnlocked,
      this.approvalService.onUnlocked,
      this.zkIdentityService.onUnlocked,
    );

    this.handler.add(
      RPCInternalAction.LOCK,
      this.lockService.lock,
      this.zkIdentityService.lock,
      this.approvalService.lock,
    );
```

Reply:
Because there is no data to unlock. When integration is done it will be added.